### ISSUE TITLE
Error using Invoke-SqlQuery from script

### ISSUE THREAD

**Issue by CaptainCrewe** [mithrandyr/SimplySql#32]
Hello,

Thank you so much for providing the SimplySQL module. My initial experiments worked well and it will be a useful resource.

I am having a problem though which, since you seem to be actively maintaining the module, may be of interest. My objective here is to get some data from a MariaDB instance running on a Raspberry Pi 4 and show the output in a WPF grid. At the moment, I'm running my code from a Windows 10 Pro machine (x64) against the database on the RPi.

Here is the code for querying the database, which I am running from a script:

`Import-Module -Name SimplySql`
`$sql = "SELECT * FROM Brokers;"`
`Open-MySqlConnection -Server '192.168.100.4' -Database 'Finances' -UserName 'sebastian' -Password 'XXXXX' $myData = Invoke-SqlQuery -Query $sql` 
`Close-SqlConnection`

[I'm aware that passing the username/password is deprecated, but it's a lot easier for troubleshooting.]

Running that results in an error:

Exception calling "Fill" with "1" argument(s): "You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near 'SELECT * FROM Brokers' at line 1" At C:\Program Files\WindowsPowerShell\Modules\SimplySql\1.6.2\Providers\MySql\provider.ps1:42 char:13 + $da.Fill($ds) + ~~~~~~~~~~~~~ + CategoryInfo : NotSpecified: (:) [], MethodInvocationException + FullyQualifiedErrorId : MySqlException

The curious thing is that if I run each line one by one in the PowerShell console, then it all works fine. I only get an error when running the code as a script.

I've tried another module, MySQL by Adam Bertram (https://mcpmag.com/articles/2016/03/02/querying-mysql-databases.aspx). That does work in script form. Initially I had the same error as above with the MySQL module until I put the $sql assignment in double-quotes. Using single-quotes resulted in the error.

With SimplySQL though the error occurs despite the $sql assignment being in double-quotes, as per the code above. It doesn't make any difference if I don't use a variable and just have the double-quoted string as an argument to Invoke-SqlQuery.

The behaviour is similar whether using PowerShell 5 or 7.  For PowerShell 7, the error occurs using the script.  It also occurs if, when entering the commands line by line, I use the variable in the Invoke-SqlQuery line (`$myData = Invoke-SqlQuery -Query $sql`) but it works fine with `$myData = Invoke-SqlQuery -Query "SELECT * FROM Brokers"`.

Using the Debugger in the PowerShell 5 ISE (with ISESteroids), I can step through and see that all is working until Invoke-SqlQuery.ps1 calls GetDataSet on line 75.  That then calls into provider.ps1 and the exception is thrown on line 42 therein.

I hope this will give enough detail to help you troubleshoot.  Please let me know if more would be helpful.

Thank you

**Comment by CaptainCrewe** [comment]
Hello again,

By chance, I noticed a problem in the script when viewing it in the file preview window that the excellent DirectoryOpus tool offers (a replacement for Windows Explorer).  That showed a space in the SELECT statement (S ELECT).

Opening the script in the PowerShell 5 ISE didn't reveal this, nor did Notepad.  So I copied my script over to the RPi and checked it there.  The SELECT looked good when opened using a local text editor.  Still failed to run though, so I edited the file locally on the RPi (using nano), deleting the entire line with the SELECT and re-composing it.  Then the script ran without error, whether -Query was passed the SELECT as a string or as a variable.  Single quotes also worked.

So there's some unhappiness in the text encoding of the script file.  This might a consequence of using the PowerShell ISE of old.  Not sure.  The good news is that SimplySql is doing a great job; the bad is that hidden encoding gremlins can thwart the development of a useful script.  I'll try making the same in Visual Studio Code and/or JetBrains Rider and see what happens.

Thank you again for providing such a useful module.
